node-test/src/test/scala/org/bitcoins/node/networking/MaxConnectionsTest.scala

```diff
@@ -0,0 +1,81 @@
+package org.bitcoins.node.networking
+
+import org.bitcoins.asyncutil.AsyncUtil
+import org.bitcoins.asyncutil.AsyncUtil.RpcRetryException
+import org.bitcoins.node.models.Peer
+import org.bitcoins.node.networking.peer.PeerHandler
+import org.bitcoins.rpc.client.common.BitcoindVersion
+import org.bitcoins.rpc.config.BitcoindInstance
+import org.bitcoins.testkit.node.{
+  CachedBitcoinSAppConfig,
+  NodeTestUtil,
+  NodeUnitTest
+}
+import org.bitcoins.testkit.rpc.BitcoindRpcTestUtil
+import org.bitcoins.testkit.util.{AkkaUtil, BitcoindRpcTest}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class MaxConnectionsTest extends BitcoindRpcTest with CachedBitcoinSAppConfig {
```

Comment:
Do we need a special test for it? How is this case  different from the case when the peer simply restarted in the middle of handshake or sending `Block` message ?

Reply:
We always should have test cases for behavior that is causing a bad user experience for our users. How will we tell if we have a regression in the future? 

>How is this case different from the case when the peer simply restarted in the middle of handshake

I'm not sure if it is different, but i know for a fact this is a pain point our users are going through, which is why i'm trying to address this specific case.